src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1770,6 +1771,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
             if (notification.HandledResponse) {
                 return;
             }
+            
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
```

Comment:
> Note: even extracted from the ticket, the client_id property is not appropriate to determine whether an application was fully authenticated (don't forget that tokens issued by the authorization endpoint will have this property).

Oh, right. I hadn't thought of that. I was really just trying to get the logic right as discussed.

> There's actually another property you can use: `OpenIdConnectConstants.Extra.ClientAuthenticated`.
I was unaware that this existed, but that works then I suppose.

> Another remark: all these checks come far too late. Ideally, they shouldn't be called after the ValidationEndpoint event. You should move this whole block at line 1732.

Ah. I assumed we would want implementers of the `ValidationEndpoint` to be able to modify the incoming data for the `ValidationEndpointNotification` if they needed to do some custom processing to change things if they didn't want to allow the default logic. If that's not the case, I can make the necessary fixes and re-squash.